Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ValueListGroup/ByValueListGroup.tsx

```diff
@@ -20,176 +20,90 @@
 import ValueListGroupForm from './ValueListGroupForm';
+import GenericByPage from '../CommonComponents/GenericByPage';
 
+const controllerPath = `${homePath}api/ValueListGroup`;
+const fieldCols: localSC.IByCol<localSC.ValueListGroupView>[] = [
+    { Field: 'Name', Label: 'Name', Type: 'string', Width: '15%' },
+    { Field: 'Description', Label: 'Description', Type: 'string', Width: 'auto' },
+    { Field: 'ItemCount', Label: 'Items', Type: 'integer', Width: '10%' }
+];
+const emptyRecord = { ID: 0, Name: '', Description: '', ItemCount: 0 };
 
-const ValueListGroups: Application.Types.iByComponent = (props) => {
-    const dispatch = useAppDispatch();
-
-    const data = useAppSelector(ValueListGroupViewSlice.SearchResults);
-    const status = useAppSelector(ValueListGroupViewSlice.SearchStatus);
-    const items = useAppSelector(ValueListSlice.Data);
-    const itemStatus = useAppSelector(ValueListSlice.Status);
-    const parentID = useAppSelector(ValueListSlice.ParentID);
-
-    const [showNew, setShowNew] = React.useState<boolean>(false);
-    const [sortField, setSortField] = React.useState<keyof localSC.ValueListGroupView>('Name');
-    const [ascending, setAscending] = React.useState<boolean>(true);
-
-    const emptyRecord = { ID: 0, Name: '', Description: '', ItemCount: 0 };
+const ValueListGroups: Application.Types.iByComponent = () => {
     let history = useHistory();
 
-    const [search, setSearch] = React.useState<Array<Search.IFilter<localSC.ValueListGroupView>>>([]);
-
+    const [refreshCount, refreshData] = React.useState<number>(0);
+    const [showNew, setShowNew] = React.useState<boolean>(false);
     const [record, setRecord] = React.useState<localSC.ValueListGroupView>(emptyRecord);
 
-    React.useEffect(() => {
-        if (status == 'unintiated' || status == 'changed')
-            dispatch(ValueListGroupViewSlice.DBSearch({ filter: search, sortField: sortField, ascending: ascending }));
-
-        return function () {
-        }
-    }, [dispatch, status]);
-
-    React.useEffect(() => {
-        dispatch(ValueListGroupViewSlice.DBSearch({ filter: search, sortField: sortField, ascending: ascending }));
-
-        return function () {
-        }
-    }, [search, sortField, ascending]);
-
-    React.useEffect(() => {
-        if (itemStatus == 'unintiated' || itemStatus == 'changed' || parentID != null)
-            dispatch(ValueListSlice.Fetch());
-    }, [itemStatus, parentID]);
-
-
+    function addValueListGroup() {
+        let handle = $.ajax({
+            type: "POST",
+            url: `${homePath}api/ValueListGroup/Add`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify(record),
+            dataType: "json",
+            cache: false,
+            async: true
+        }).done(() => {
+            refreshData(x => x + 1);
+        })
+
+        return () => {
+            if (handle != null && handle.abort != null) handle.abort();
+        };
+    }
```

Comment:
This should be replaced by generic controller, similar to above, but just FYI:

In general, we DON'T want to use cleanup functions like this in this kind of function. Imagine for some reason the server is slower than a user, and a user adds two groups quickly before the server can respond. If this cleanup function were to be executed (which wouldn't be the case here since we never do anything with it where we call this function) the JS would cancel the first call, causing the first group to never be added. Usually, we use cleanup functions for server calls that have no use if another one is made (i.e. calls to search the database for some UI component).